### DISCUSSION TITLE
vscode - disable reformat on save

### DISCUSSION THREAD

**Discussion by hwspeedy** [hardinfo2/hardinfo2#89] [General]
To be able to track the changes, we cannot reformat the entire file on simple commits. This is happening on VSCode as default on save.

Here is a howto disable reformat in VSCode depending on setup:
https://stackoverflow.com/questions/61827206/disable-all-auto-formatting-on-save-in-vscode

as always you can track your current changes before commit locally by:
git status        (overview)
git diff (Changes)
fix the unwanted diff and test again before commit and especially before push to our common base.

/hwspeedy

**Comment by topazus** [comment]
Speaking about format issue, may we can consider using the clang-format tool like other c++ projects?

[1] https://github.com/fmtlib/fmt/blob/master/.clang-format
[2] https://clang.llvm.org/docs/ClangFormat.html

**Comment by hwspeedy** [comment]
@topazus 

Probably not...

1. The coding style is KISS - Keep It Simple Stupid.
**Readability is most important factor.**
We have 4 indent, normal rules for pointer placement (* at variable), no limit on line length - but if you cannot read it on a full HD screen - it fails.
Fix most extra warnings from all compilers and improve for maximum tool coverage.
Compile language version is c89 for main program. In a couple of years, c11 will probably be fine if needed - Always 10+ years support.

2. Eventhough a clean code base is nice - autoformat brings possible errors and still needs review to ensure safe code base (Huge task) and brings the users nothing if the code is not faulty and unreadable. Hence not Lean, not KISS.

Clang, Gnu, Google - none of these coding styles cover above - but we use most of all of them.

Does that make sense or do you have some good bullet points for your use autoformat idea?
